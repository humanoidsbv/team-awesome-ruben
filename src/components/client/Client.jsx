import React from 'react';

const Client = () => <React.Fragment></React.Fragment>;

export default Client;
