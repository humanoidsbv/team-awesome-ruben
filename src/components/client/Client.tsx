import React, { useState } from 'react';

import IconArrowExpand from '../../assets/icons/icon-arrow-expand.svg';
import styles from './Client.module.css';

import { ClientInterface } from '../../redux/clients/types';

interface ClientPropsInterface extends ClientInterface {
  client: ClientPropsInterface;
}

const Client = ({
  client: { companyName }
}: ClientPropsInterface): React.ReactFragment => {
  const [isExpanded, setIsExpanded] = useState(false);

  const handleClick = () => setIsExpanded(!isExpanded);

  //* mock data for now!
  const clientNumber = 'CLN_';
  const projects = ['project1'];
  const activeTeamMembers = ['Antje Adriaens'];

  return (
    <button
      className={`${styles.container}
        ${isExpanded && styles.containerActive}`}
      onClick={handleClick}
      type="button"
    >
      <div className={styles.nameContainer}>
        <span>{companyName}</span>
      </div>
      <IconArrowExpand
        className={`${styles.arrowExpand}
          ${isExpanded && styles.arrowExpandActive}`}
      />
      <span
        className={`${styles.descriptionBlock}
          ${isExpanded && styles.descriptionBlockActive}`}
      >
        {`Detailed information about ${companyName}`}
      </span>
      <div
        className={`
          ${styles.detailContainer}
          ${isExpanded && styles.detailContainerActive}`}
      >
        <span className={styles.detail}>{clientNumber}</span>
        <span className={styles.detailDescription}>Client number</span>
      </div>

      <div
        className={`
          ${styles.detailContainer}
          ${isExpanded && styles.detailContainerActive}`}
      >
        <span className={styles.detail}>
          {String(projects.length).padStart(3, '0')}
        </span>
        <span className={styles.detailDescription}>Active projects</span>
      </div>

      <div
        className={`
          ${styles.detailContainer}
          ${isExpanded && styles.detailContainerActive}`}
      >
        <span className={styles.detail}>
          {String(activeTeamMembers.length).padStart(3, '0')}
        </span>
        <span className={styles.detailDescription}>Assigned team</span>
      </div>

      <div
        className={`
          ${styles.detailContainer} ${styles.fullWidth}
          ${isExpanded && styles.detailContainerActive}`}
      >
        <span className={`${styles.detail}`}>
          {String(activeTeamMembers.length).padStart(3, '0')}
        </span>
        <span className={styles.detailDescription}>Assigned</span>
      </div>
      <div
        className={`
          ${styles.detailContainer} ${styles.fullWidth}
          ${isExpanded && styles.detailContainerActive}`}
      >
        <span className={styles.detail}>
          {String(activeTeamMembers.length).padStart(3, '0')}
        </span>
        <span className={styles.detailDescription}>Assigned</span>
      </div>
    </button>
  );
};

export default Client;
