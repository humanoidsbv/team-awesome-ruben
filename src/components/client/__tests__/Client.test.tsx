import * as React from 'react';
import { shallow } from 'enzyme';

import Client from '../Client';

const mockedClientsData = {
  branch: '',
  branchAddress: '',
  companyName: '',
  id: 0,
  locality: '',
  postalCode: ''
};

it('matches component output with expected output', () => {
  const client = shallow(<Client client={mockedClientsData} />);
  expect(client).toMatchSnapshot();
});
