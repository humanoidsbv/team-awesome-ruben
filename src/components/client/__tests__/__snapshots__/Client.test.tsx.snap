// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<button
  className="container
        false"
  onClick={[Function]}
  type="button"
>
  <div
    className="nameContainer"
  >
    <span />
  </div>
  <test-file-stub
    className="arrowExpand
          false"
  />
  <span
    className="descriptionBlock
          false"
  >
    Detailed information about 
  </span>
  <div
    className="
          detailContainer
          false"
  >
    <span
      className="detail"
    >
      CLN_
    </span>
    <span
      className="detailDescription"
    >
      Client number
    </span>
  </div>
  <div
    className="
          detailContainer
          false"
  >
    <span
      className="detail"
    >
      001
    </span>
    <span
      className="detailDescription"
    >
      Active projects
    </span>
  </div>
  <div
    className="
          detailContainer
          false"
  >
    <span
      className="detail"
    >
      001
    </span>
    <span
      className="detailDescription"
    >
      Assigned team
    </span>
  </div>
  <div
    className="
          detailContainer fullWidth
          false"
  >
    <span
      className="detail"
    >
      001
    </span>
    <span
      className="detailDescription"
    >
      Assigned
    </span>
  </div>
  <div
    className="
          detailContainer fullWidth
          false"
  >
    <span
      className="detail"
    >
      001
    </span>
    <span
      className="detailDescription"
    >
      Assigned
    </span>
  </div>
</button>
`;
